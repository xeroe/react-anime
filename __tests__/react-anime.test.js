import React from 'react';
import { renderToString } from 'react-dom';
import renderer from 'react-test-renderer';
import Anime from '../src/react-anime';

const reactAnimationFrame = (function() {
    var lastTime = 0;
    var vendors = ['ms', 'moz', 'webkit', 'o'];
    for(var x = 0; x < vendors.length && !window.requestAnimationFrame; ++x) {
        window.requestAnimationFrame = window[vendors[x]+'RequestAnimationFrame'];
        window.cancelAnimationFrame = window[vendors[x]+'CancelAnimationFrame']
                                   || window[vendors[x]+'CancelRequestAnimationFrame'];
    }


    if (!window.requestAnimationFrame)
        window.requestAnimationFrame = function(callback, element) {
            var currTime = new Date().getTime();
            var timeToCall = Math.max(0, 16 - (currTime - lastTime));
            var id = window.setTimeout(function() { callback(currTime + timeToCall); },
              timeToCall);
            lastTime = currTime + timeToCall;
            return id;
        };

    if (!window.cancelAnimationFrame)
        window.cancelAnimationFrame = function(id) {
            clearTimeout(id);
        };
}());

//snapshot test
it('2 children delayed fade in', () => {
  const tree = renderer.create(
    <Anime
      easing="easeOutElastic"
      loop={true}
      duration={1000}
      direction="alternate"
      loop={true}
      delay={(_, index) => index * 240}
    >
      <p>1</p>
      <p>2</p>
    </Anime>,
    {
      createNodeMock: _ => ({ nodeType: false }),
    },
  );

  expect(tree.toJSON()).toMatchSnapshot();
});
