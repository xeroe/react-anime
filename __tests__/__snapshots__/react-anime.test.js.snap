// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2 children delayed fade in 1`] = `
Array [
  <div>
    <p>
      1
    </p>
  </div>,
  <div>
    <p>
      2
    </p>
  </div>,
]
`;
